### PR TITLE
[TACHYON-967] Throw Thrift Exceptions in Worker

### PR DESCRIPTION
**Pull request by manugoyal**
https://tachyon.atlassian.net/browse/TACHYON-967

We add new Thrift exceptions to mirror the exceptions being used in the Worker classes (NotFoundException, InvalidStateException, etc), and then throw those across the RPC layer.

Note that in user-facing code (TachyonFS, *BlockOutStream, *BlockInStream), we don't expose these thrift exceptions, because TachyonFS is deprecated, and because the OutStream and InStream classes conform to the Java interface of only throwing IOExceptions.

### CODE REVIEW

**Comment by calvinjia** [comment]
@manugoyal Could you update the PR with the JIRA ticket?

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 1]
```diff
@@ -353,13 +382,31 @@ public synchronized void mustConnect() throws IOException {
...
+   * @throws IOException if an I/O error occurs
+   * @throws BlockDoesNotExistException if blockId cannot be found
+   * @throws BlockAlreadyExistsException if blockId already exists in committed blocks of the
+   *         newLocation
+   * @throws InvalidWorkerStateException if blockId has not been committed
+   * @throws WorkerOutOfSpaceException if newLocation does not have enough extra space to hold the
+   *         block
    */
-  public synchronized boolean promoteBlock(long blockId) throws IOException {
+  public synchronized boolean promoteBlock(long blockId) throws IOException, ThriftIOException,
```
Comment (yupeng9) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: Missing doc for ThriftIOException?

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 2]
```diff
@@ -372,18 +419,31 @@ public synchronized boolean promoteBlock(long blockId) throws IOException {
    * @param blockId The id of the block
    * @param initialBytes The initial size bytes allocated for the block
    * @return the temporary path of the block
-   * @throws IOException
+   * @throws IOException if an I/O error occurs
```
Comment (yupeng9) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: ditto

File: servers/src/main/java/tachyon/worker/block/BlockDataManager.java [hunk 3]
```diff
@@ -223,20 +225,23 @@ public void commitBlock(long sessionId, long blockId) throws AlreadyExistsExcept
...
+   * @throws BlockDoesNotExistException if blocks in eviction plan can not be found
    * @throws IOException if blocks in eviction plan fail to be moved or deleted
-   * @throws InvalidStateException if blocks to be moved/deleted in eviction plan is uncommitted
+   * @throws InvalidWorkerStateException if blocks to be moved/deleted in eviction plan is
+   *         uncommitted
    */
-  // TODO: exceptions like NotFoundException, IOException and InvalidStateException here involves
-  // implementation details, also, AlreadyExistsException has two possible semantic now, these are
+  // TODO: exceptions like BlockDoesNotExistException, IOException and InvalidWorkerStateException
+  // here involves
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/worker/block/BlockDataManager.java]: The line breaks early after auto-formatting

File: servers/src/main/java/tachyon/worker/block/BlockDataManager.java [hunk 4]
```diff
@@ -253,20 +258,23 @@ public String createBlock(long sessionId, long blockId, int tierAlias, long init
...
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/worker/block/BlockDataManager.java]: ditto

File: servers/src/main/java/tachyon/worker/block/BlockDataManager.java [hunk 5]
```diff
@@ -430,23 +441,25 @@ public void removeBlock(long sessionId, long blockId) throws InvalidStateExcepti
...
    *         location
-   * @throws InvalidStateException if the space requested is less than current space or blocks to
-   *         move/evict in {@link tachyon.worker.block.evictor.EvictionPlan} is uncommitted
+   * @throws InvalidWorkerStateException if the space requested is less than current space or blocks
+   *         to move/evict in {@link tachyon.worker.block.evictor.EvictionPlan} is uncommitted
    */
-  // TODO: exceptions like IOException AlreadyExistsException and InvalidStateException here
-  // involves implementation details, also, NotFoundException has two semantic now, revisit this
+  // TODO: exceptions like IOException BlockAlreadyExistsException and InvalidWorkerStateException
+  // here
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/worker/block/BlockDataManager.java]: here

**Comment by yupeng9** [comment]
Thanks for the big improvement on the exception structure! It overall looks good and I have some minor comments.

Also please resolve the conflicts.

**Comment by manugoyal** [comment]
Can we re-run the tests, I think they're passing on my machine.

**Comment by yupeng9** [comment]
Jenkins, test this please

**Comment by yupeng9** [comment]
LGTM

**Comment by manugoyal** [comment]
Jenkins, test this please

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 6]
```diff
@@ -205,7 +208,11 @@ private TachyonFS(TachyonConf tachyonConf) {
    */
   synchronized void accessLocalBlock(long blockId) throws IOException {
     if (mWorkerClient.isLocal()) {
-      mWorkerClient.accessBlock(blockId);
+      try {
+        mWorkerClient.accessBlock(blockId);
+      } catch (BlockDoesNotExistException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: Should this just be ``TException e`?

That way if the implementation of accessBlock changes potentially throwing a different set of Thrift exceptions, we do not have to change this.

This applies to all locations where we convert a Thrift exception to an IO exception.

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 7]
```diff
@@ -216,7 +223,13 @@ synchronized void accessLocalBlock(long blockId) throws IOException {
    * @throws IOException if the underlying worker RPC fails
    */
   synchronized void addCheckpoint(long fid) throws IOException {
-    mWorkerClient.addCheckpoint(fid);
+    try {
+      mWorkerClient.addCheckpoint(fid);
+    } catch (FailedToCheckpointException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: As per my previous comment, this should just be ``TException e`` and there should only be one catch clause.

Alternatively, if we choose to list all exception cases, we should do so consistently (e.g. requestBlockLocation() below should list all of its exceptions and not just ``TException``).

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 8]
```diff
@@ -855,7 +881,12 @@ synchronized String lockBlock(long blockId, int blockLockId) throws IOException
     if (!mWorkerClient.isLocal()) {
       return null;
     }
-    String blockPath = mWorkerClient.lockBlock(blockId);
+    String blockPath = null;
+    try {
+      blockPath = mWorkerClient.lockBlock(blockId);
+    } catch (InvalidWorkerStateException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: TException

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 9]
```diff
@@ -1076,7 +1111,11 @@ synchronized boolean unlockBlock(long blockId, int blockLockId) throws IOExcepti
 
     mLockedBlockIdToPath.remove(blockId);
     mLockedBlockIds.remove(blockId);
-    return mWorkerClient.unlockBlock(blockId);
+    try {
+      return mWorkerClient.unlockBlock(blockId);
+    } catch (BlockDoesNotExistException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: TException

File: clients/unshaded/src/main/java/tachyon/client/block/LocalBlockInStream.java [hunk 10]
```diff
@@ -53,7 +55,12 @@ public LocalBlockInStream(long blockId) throws IOException {
     mContext = BlockStoreContext.INSTANCE;
     mWorkerClient =
         mContext.acquireWorkerClient(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
-    String blockPath = mWorkerClient.lockBlock(blockId);
+    String blockPath = null;
+    try {
+      blockPath = mWorkerClient.lockBlock(blockId);
+    } catch (InvalidWorkerStateException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/block/LocalBlockInStream.java]: TException

File: clients/unshaded/src/main/java/tachyon/client/block/LocalBlockInStream.java [hunk 11]
```diff
@@ -78,7 +85,11 @@ public void close() throws IOException {
     if (mClosed) {
       return;
     }
-    mWorkerClient.unlockBlock(mBlockId);
+    try {
+      mWorkerClient.unlockBlock(mBlockId);
+    } catch (BlockDoesNotExistException e) {
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/block/LocalBlockInStream.java]: TException

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 12]
```diff
@@ -99,13 +99,17 @@ public WorkerClient(NetAddress workerNetAddress, ExecutorService executorService
...
    */
-  public synchronized void accessBlock(long blockId) throws IOException {
+  public synchronized void accessBlock(long blockId) throws IOException,
+          BlockDoesNotExistException {
     mustConnect();
 
     try {
       mClient.accessBlock(blockId);
+    } catch (BlockDoesNotExistException e) {
```
Comment (jsimsa) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: Do we need these catch clauses if we are just rethrowing the exception? This applies to multiple locations in the WorkerClient class.

File: common/src/thrift/tachyon.thrift [hunk 13]
```diff
@@ -109,10 +109,18 @@ exception FileAlreadyExistException {
   1: string message
 }
 
+exception BlockAlreadyExistsException {
```
Comment (jsimsa) [inline common/src/thrift/tachyon.thrift]: I think we should put the exceptions into a separate thrift file, keep them alphabetically sorted, and add a comment that explains their purpose? It is fine to do this in a separate PR, just add a TODO (and a JIRA ticket) for this. Thanks!

File: integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java [hunk 14]
```diff
@@ -275,7 +275,7 @@ public void requestSpaceTest() throws Exception {
     Exception exception = null;
     try {
       mWorkerServiceHandler.requestBlockLocation(SESSION_ID, blockId2, WORKER_CAPACITY_BYTES + 1);
-    } catch (OutOfSpaceException oose) {
+    } catch (WorkerOutOfSpaceException oose) {
```
Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/worker/BlockServiceHandlerIntegrationTest.java]: Declare the following public member for the test class:

```@Rule
  public final ExpectedException mException = ExpectedException.none();```

and then use ``mException.expect(WorkerOutOfSpaceException.class)`` here instead of the try / catch idiom.

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 15]
```diff
@@ -92,7 +92,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
         getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
             response);
         return;
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: should the exception variable name change to bdnee? 

(I personally do not see the reason for using these verbose variable names and would simply use e everywhere.)

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 16]
```diff
@@ -136,7 +136,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       request.setAttribute("fatalError", iae.getLocalizedMessage());
       getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
       return;
-    } catch (NotFoundException nfe) {
+    } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: ditto

File: servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java [hunk 17]
```diff
@@ -204,7 +204,8 @@ private UiFileInfo getUiFileInfo(TachyonFileSystem tachyonFileSystem, TachyonURI
    * @throws IOException
    */
   private UiFileInfo getUiFileInfo(TachyonFileSystem tachyonFileSystem, long fileId,
-      TachyonURI filePath) throws FileDoesNotExistException, NotFoundException, IOException {
+      TachyonURI filePath) throws FileDoesNotExistException, BlockDoesNotExistException,
+      IOException {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java]: why is this on a new line?

File: servers/src/main/java/tachyon/worker/block/BlockLockManager.java [hunk 18]
```diff
@@ -133,14 +134,15 @@ public void unlockBlock(long lockId) throws NotFoundException {
     lock.unlock();
   }
 
-  // TODO(bin): Temporary, remove me later.
-  public void unlockBlock(long sessionId, long blockId) throws NotFoundException {
+  // TODO(bin): temporary, remove me later.
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockLockManager.java]: The style is either to capitalize sentences.

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java [hunk 19]
```diff
@@ -69,18 +69,18 @@ public static BlockMetadataManager newBlockMetadataManager() {
       ret.initBlockMetadataManager();
       // caller of newBlockMetadataManager should not be forced to catch and handle these exceptions
       // since it is the responsibility of BlockMetadataManager.
-    } catch (AlreadyExistsException aee) {
+    } catch (BlockAlreadyExistsException aee) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: fix variable name

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java [hunk 20]
```diff
@@ -69,18 +69,18 @@ public static BlockMetadataManager newBlockMetadataManager() {
...
       throw new RuntimeException(aee);
     } catch (IOException ioe) {
       throw new RuntimeException(ioe);
-    } catch (OutOfSpaceException ooe) {
+    } catch (WorkerOutOfSpaceException ooe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: ditto

File: servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java [hunk 21]
```diff
@@ -155,20 +152,19 @@ public String getSessionUfsTempFolder(long sessionId) {
...
-  public String lockBlock(long blockId, long sessionId)
-      throws FileDoesNotExistException, TException {
+  public String lockBlock(long blockId, long sessionId) throws FileDoesNotExistException,
+      InvalidWorkerStateException {
     try {
       long lockId = mWorker.lockBlock(sessionId, blockId);
       return mWorker.readBlock(sessionId, blockId, lockId);
-    } catch (NotFoundException nfe) {
-      // TODO(cc): Reconsider this, maybe it is because lockId can not be found.
+    } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java]: fix variable name

File: servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java [hunk 22]
```diff
@@ -155,20 +152,19 @@ public String getSessionUfsTempFolder(long sessionId) {
...
+      // TODO(cc): reconsider this, maybe it is because lockId can not be found
       throw new FileDoesNotExistException(nfe.getMessage());
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java]: wouldn't ``BlockDoesNotExistException`` make more sense?

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 23]
```diff
@@ -464,7 +473,7 @@ private void abortBlockInternal(long sessionId, long blockId) throws NotFoundExc
       mMetadataWriteLock.lock();
       try {
         mMetaManager.abortTempBlockMeta(tempBlockMeta);
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: fix variable name

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 24]
```diff
@@ -513,11 +523,11 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId)
       mMetadataWriteLock.lock();
       try {
         mMetaManager.commitTempBlockMeta(tempBlockMeta);
-      } catch (AlreadyExistsException aee) {
+      } catch (BlockAlreadyExistsException aee) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 25]
```diff
@@ -513,11 +523,11 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId)
...
         throw Throwables.propagate(aee); // we shall never reach here
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 26]
```diff
@@ -513,11 +523,11 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId)
...
         throw Throwables.propagate(nfe); // we shall never reach here
-      } catch (OutOfSpaceException ose) {
+      } catch (WorkerOutOfSpaceException ose) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 27]
```diff
@@ -564,12 +574,12 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
         // Add allocated temp block to metadata manager. This should never fail if allocator
         // correctly assigns a StorageDir.
         mMetaManager.addTempBlockMeta(tempBlock);
-      } catch (OutOfSpaceException ose) {
+      } catch (WorkerOutOfSpaceException ose) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 28]
```diff
@@ -564,12 +574,12 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
...
         // If we reach here, allocator is not working properly
         LOG.error("Unexpected failure: " + initialBlockSize + " bytes allocated at " + location
             + " by allocator, but addTempBlockMeta failed");
         throw Throwables.propagate(ose);
-      } catch (AlreadyExistsException aee) {
+      } catch (BlockAlreadyExistsException aee) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 29]
```diff
@@ -602,9 +612,9 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
       }
       // Increase the size of this temp block
       try {
-        mMetaManager.resizeTempBlockMeta(tempBlockMeta,
-            tempBlockMeta.getBlockSize() + additionalBytes);
-      } catch (InvalidStateException ise) {
+        mMetaManager.resizeTempBlockMeta(tempBlockMeta, tempBlockMeta.getBlockSize()
+            + additionalBytes);
+      } catch (InvalidWorkerStateException ise) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 30]
```diff
@@ -641,11 +652,11 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
     for (Pair<Long, BlockStoreLocation> blockInfo : plan.toEvict()) {
       try {
         removeBlockInternal(sessionId, blockInfo.getFirst(), blockInfo.getSecond());
-      } catch (InvalidStateException ise) {
+      } catch (InvalidWorkerStateException ise) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 31]
```diff
@@ -641,11 +652,11 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
...
         // Evictor is not working properly
         LOG.error("Failed to evict blockId " + blockInfo.getFirst() + ", this is temp block");
         continue;
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 32]
```diff
@@ -680,13 +691,13 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
         MoveBlockResult moveResult;
         try {
           moveResult = moveBlockInternal(sessionId, blockId, oldLocation, newLocation);
-        } catch (InvalidStateException ise) {
+        } catch (InvalidWorkerStateException ise) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 33]
```diff
@@ -680,13 +691,13 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
...
           // Evictor is not working properly
           LOG.error("Failed to evict blockId " + blockId + ", this is temp block");
           continue;
-        } catch (AlreadyExistsException aee) {
+        } catch (BlockAlreadyExistsException aee) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 34]
```diff
@@ -680,13 +691,13 @@ private void freeSpaceInternal(long sessionId, long availableBytes, BlockStoreLo
...
           continue;
-        } catch (NotFoundException nfe) {
+        } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 35]
```diff
@@ -786,11 +799,11 @@ private MoveBlockResult moveBlockInternal(long sessionId, long blockId,
         // If this metadata update fails, we panic for now.
         // TODO(bin): Implement rollback scheme to recover from IO failures.
         mMetaManager.moveBlockMeta(srcBlockMeta, dstTempBlock);
-      } catch (AlreadyExistsException aee) {
+      } catch (BlockAlreadyExistsException aee) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 36]
```diff
@@ -786,11 +799,11 @@ private MoveBlockResult moveBlockInternal(long sessionId, long blockId,
...
         throw Throwables.propagate(aee); // we shall never reach here
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 37]
```diff
@@ -786,11 +799,11 @@ private MoveBlockResult moveBlockInternal(long sessionId, long blockId,
...
         throw Throwables.propagate(nfe); // we shall never reach here
-      } catch (OutOfSpaceException ose) {
+      } catch (WorkerOutOfSpaceException ose) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 38]
```diff
@@ -841,7 +855,7 @@ private void removeBlockInternal(long sessionId, long blockId, BlockStoreLocatio
       mMetadataWriteLock.lock();
       try {
         mMetaManager.removeBlockMeta(blockMeta);
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: ditto

File: servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java [hunk 39]
```diff
@@ -97,7 +97,7 @@ protected StorageDirView cascadingEvict(long bytesToBeAvailable, BlockStoreLocat
                 block.getBlockSize());
           }
         }
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java]: fix variable name

File: servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java [hunk 40]
```diff
@@ -125,7 +125,7 @@ protected StorageDirView cascadingEvict(long bytesToBeAvailable, BlockStoreLocat
             plan.toEvict().add(new Pair<Long, BlockStoreLocation>(blockId,
                 candidateDirView.toBlockStoreLocation()));
           }
-        } catch (NotFoundException nfe) {
+        } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java]: ditto

File: servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java [hunk 41]
```diff
@@ -155,7 +155,7 @@ protected StorageDirView cascadingEvict(long bytesToBeAvailable, BlockStoreLocat
               nextDirView.toBlockStoreLocation()));
           candidateDirView.markBlockMoveOut(blockId, block.getBlockSize());
           nextDirView.markBlockMoveIn(blockId, block.getBlockSize());
-        } catch (NotFoundException nfe) {
+        } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/evictor/EvictorBase.java]: ditto

File: servers/src/main/java/tachyon/worker/netty/DataServerHandler.java [hunk 42]
```diff
@@ -102,7 +102,7 @@ private void handleBlockReadRequest(final ChannelHandlerContext ctx,
     long lockId;
     try {
       lockId = mDataManager.lockBlock(Sessions.DATASERVER_SESSION_ID, blockId);
-    } catch (NotFoundException ioe) {
+    } catch (BlockDoesNotExistException ioe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/netty/DataServerHandler.java]: fix variable name

File: servers/src/main/java/tachyon/worker/netty/DataServerHandler.java [hunk 43]
```diff
@@ -114,9 +114,9 @@ private void handleBlockReadRequest(final ChannelHandlerContext ctx,
     BlockReader reader;
     try {
       reader = mDataManager.readBlockRemote(Sessions.DATASERVER_SESSION_ID, blockId, lockId);
-    } catch (NotFoundException nfe) {
+    } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/netty/DataServerHandler.java]: ditto

File: servers/src/main/java/tachyon/worker/netty/DataServerHandler.java [hunk 44]
```diff
@@ -114,9 +114,9 @@ private void handleBlockReadRequest(final ChannelHandlerContext ctx,
...
       throw new IOException(nfe);
-    } catch (InvalidStateException fpe) {
+    } catch (InvalidWorkerStateException fpe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/netty/DataServerHandler.java]: ditto

File: servers/src/main/java/tachyon/worker/netty/DataServerHandler.java [hunk 45]
```diff
@@ -143,7 +143,7 @@ private void handleBlockReadRequest(final ChannelHandlerContext ctx,
     } finally {
       try {
         mDataManager.unlockBlock(lockId);
-      } catch (NotFoundException nfe) {
+      } catch (BlockDoesNotExistException nfe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/netty/DataServerHandler.java]: ditto

File: servers/src/main/java/tachyon/worker/nio/NIODataServer.java [hunk 46]
```diff
@@ -313,7 +313,7 @@ private void write(SelectionKey key) {
       // TODO(calvin): Reconsider how we handle this exception.
       try {
         mDataManager.unlockBlock(sendMessage.getLockId());
-      } catch (NotFoundException ioe) {
+      } catch (BlockDoesNotExistException ioe) {
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/nio/NIODataServer.java]: fix variable name

**Comment by jsimsa** [comment]
Manu, thank you very much for this PR. It greatly improves the consistency with which Tachyon handles exception.

Sorry for the laundry list of comments. One comment that repeats often is to fix variable mismatching variable names of exceptions. I suggest that we use ``e`` everywhere (instead of the current abbreviation convention) so that future renaming won't have this problem.

**Comment by jsimsa** [comment]
Also, please resolve the merge conflicts.

**Comment by manugoyal** [comment]
Seems like there's some alternative approaches being discussed on the doc. Should we wait till that's figured out before moving forward on this?

**Comment by jsimsa** [comment]
Makes sense. We can chat more about this in person on Friday.

**Comment by manugoyal** [comment]
So I guess we can close this?
